yamllint/rules/line_length.py

```diff
@@ -140,7 +140,11 @@ def check(conf, line):
                 start += 1
 
             if start != line.end:
-                if line.buffer[start] in ('#', '-'):
+                if line.buffer[start] in ('#'):
+                    idx = line.buffer.find(' ', start, line.end)
```

Comment:
I don't like this approach because it doesn't do what your commit title says: `skip all hash signs starting comment`. The current implementation allows **any** non-space character after `#`, which includes chars like `A` or `\n` or `🤙`.

Maybe something like:
```python
if line.buffer[start] == '#':
    while line.buffer[start] == '#':
         start += 1
    start += 1
elif line.buffer[start] == '-':
```
?

Reply:
Done, please check